connectors/sources/mysql.py

```diff
@@ -224,16 +218,27 @@ async def ping(self):
             if self.ssl_enabled
             else None,
         }
-        logger.info("Pinging MySQL...")
-        if self.connection_pool is None:
-            self.connection_pool = await aiomysql.create_pool(**connection_string)
+
         try:
```

Comment:
A more correct pattern is:

```
try:
    connection_pool = await aiomysql.create_pool(**connection_string)
except Exception as e:
    # deal with errors creating it and exit
    XXX

try:
    yield connection_pool
finally:
    connection_pool.close()
    await connection_pool.wait_closed()
```

so you don't mix issues with pool creation and pool usage
    
```

Reply:
Totally missed that, thanks!